robot_smach_states/src/robot_smach_states/human_interaction/human_interaction.py

```diff
@@ -414,6 +414,51 @@ def learn_person_procedure(robot, person_name="", n_samples=5, timeout=5.0):
 
 ##########################################################################################################################################
 
+class AskPersonName(smach.State):
+    """
+    Ask the person's name, and try to hear one of the given names
+    """
+    def __init__(self, robot, personNameDes, name_options, defaultName='Operator'):
+        smach.State.__init__(   self, outcomes=['succeeded', 'failed'])
+
+        self.robot = robot
+        self.personNameDes = personNameDes
+        self.defaultName = defaultName
+        self.name_options = name_options
+
+    def execute(self, userdata=None):
+        rospy.loginfo("AskPersonName")
+
+        self.robot.speech.speak("What is your name?", block=True)
+
+        names_spec = "T['name': N] -> NAME[N]\n\n"
+        for dn in self.name_options:
+            names_spec += "NAME['{name}'] -> {name}\n".format(name=dn)
+        spec = ds.Designator(names_spec)
+
+        answer = ds.VariableDesignator(resolve_type=QueryResult)
+        state = HearOptionsExtra(self.robot, spec, answer.writeable)
+        outcome = state.execute()
+
+        if not outcome == "heard":
+            self.personNameDes.write(self.defaultName)
+
+            rospy.logwarn("Speech recognition outcome was not successful (outcome: '{0}'). Using default name '{1}'".format(str(outcome), self.personNameDes.resolve()))
+            return 'failed'
+        else:
+            try:
+                print answer.resolve()
+                name = answer.resolve().semantics["name"]
+                self.personNameDes.write(name)
+
+                rospy.loginfo("Result received from speech recognition is '" + name + "'")
+            except KeyError, ke:
+                rospy.loginfo("KeyError resolving the name heard: " + str(ke))
```

Comment:
Why catch this silently? Your challenge will crash later on and you'll have to find this line that was printed (as info) a while ago. If an exception occurs: have it fail.